src/index.js

```diff
@@ -4,52 +4,70 @@ import ReactDOMServer from 'react-dom/server';
 import hypernova, { serialize, load, toScript, fromScript } from 'hypernova';
 import { StyleSheet, StyleSheetServer } from 'aphrodite';
 
-export const renderReactWithAphrodite = (name, component) => hypernova({
-  server() {
-    return (props) => {
-      const { html, css } = StyleSheetServer.renderStatic(() => (
-        ReactDOMServer.renderToString(React.createElement(component, props))
-      ));
-
-      const style = `<style data-aphrodite="data-aphrodite">${css.content}</style>`;
-      const markup = serialize(name, html, props);
-      const classNames = toScript({ 'aphrodite-css': name }, css.renderedClassNames);
-
-      return `${style}\n${markup}\n${classNames}`;
-    };
-  },
-
-  client() {
-    const classNames = fromScript({ 'aphrodite-css': name });
-    if (classNames) StyleSheet.rehydrate(classNames);
-
-    const payloads = load(name);
-    if (payloads) {
-      payloads.forEach((payload) => {
-        const { node, data } = payload;
-        if (node) {
-          const element = React.createElement(component, data);
-          ReactDOM.render(element, node);
-        }
-      });
-    }
-
-    return component;
-  },
-});
-
-export const renderReactWithAphroditeStatic = (name, component) => hypernova({
-  server() {
-    return (props) => {
-      const { html, css } = StyleSheetServer.renderStatic(() => (
-        ReactDOMServer.renderToStaticMarkup(React.createElement(component, props))
-      ));
-
-      const style = `<style data-aphrodite="data-aphrodite">${css.content}</style>`;
-
-      return `${style}\n${html}`;
-    };
-  },
-
-  client() {},
-});
+const config = {
+  enhancers: [], // HOCs applied to the component before render: e0(e1(e2(...en(component))))
+};
+
+export const setRenderEnhancers = (...enhancers) => {
+  config.enhancers = enhancers.filter(hoc => typeof hoc === 'function'); // clear & set
+};
+
+const enhance = component => config.enhancers.reverse().reduce((x, f) => f(x), component);
```

Comment:
`.reverse()` is a mutator method; every time this is invoked, it will flip the order of `config.enhancers`. You may want to use `reduceRight` here, instead

Reply:
Good call!